Problem: cmake build is broken

I am trying to compile with cmake. I did an "mkdir build && cd build && cmake ..", and got this error:

zoobab@chuchu /home/zoobab/soft/zproject [545]$ mkdir build
zoobab@chuchu /home/zoobab/soft/zproject [546]$ cd build/
zoobab@chuchu /home/zoobab/soft/zproject/build [547]$ cmake ..
-- The C compiler identification is GNU 4.9.3
-- The CXX compiler identification is GNU 4.9.3
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:21 (file):
  file STRINGS file "/home/zoobab/soft/zproject/include/myproject_library.h"
  cannot be read.


CMake Error at CMakeLists.txt:24 (message):
  failed to parse MYPROJECT_VERSION_MAJOR from myproject.h


-- Configuring incomplete, errors occurred!
See also "/home/zoobab/soft/zproject/build/CMakeFiles/CMakeOutput.log".

--- a/.gitignore
+++ b/.gitignore
@@ -16,3 +16,7 @@ libtool
 *.zip
 *.la
 *.swp
+CMakeCache.txt
+CMakeFiles/
+CTestTestfile.cmake
+cmake_install.cmake

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,20 +14,6 @@ enable_testing()
 set(SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 
 ########################################################################
-# determine version
-########################################################################
-foreach(which MAJOR MINOR PATCH)
-    file(STRINGS "${SOURCE_DIR}/include/myproject_library.h" MYPROJECT_VERSION_STRING REGEX "#define MYPROJECT_VERSION_${which}")
-    string(REGEX MATCH "#define MYPROJECT_VERSION_${which} ([0-9_]+)" MYPROJECT_REGEX_MATCH "${MYPROJECT_VERSION_STRING}")
-    if (NOT MYPROJECT_REGEX_MATCH)
-        message(FATAL_ERROR "failed to parse MYPROJECT_VERSION_${which} from myproject.h")
-    endif()
-    set(MYPROJECT_${which}_VERSION ${CMAKE_MATCH_1})
-endforeach(which)
-
-set(MYPROJECT_VERSION ${MYPROJECT_MAJOR_VERSION}.${MYPROJECT_MINOR_VERSION}.${MYPROJECT_PATCH_VERSION})
-
-########################################################################
 # platform.h
 ########################################################################
 include(CheckIncludeFile)
@@ -84,45 +70,6 @@ set (myproject_headers
 )
 source_group ("Header Files" FILES ${myproject_headers})
 install(FILES ${myproject_headers} DESTINATION include)
-
-########################################################################
-# library
-########################################################################
-include_directories("${SOURCE_DIR}/src" "${SOURCE_DIR}/include")
-set (myproject_sources
-)
-source_group ("Source Files" FILES ${myproject_sources})
-if (NOT DEFINED BUILD_SHARED_LIBS)
-    SET(BUILD_SHARED_LIBS ON)
-endif()
-add_library(myproject ${myproject_sources})
-set_target_properties(myproject PROPERTIES DEFINE_SYMBOL "LIBMYPROJECT_EXPORTS")
-set_target_properties(myproject PROPERTIES LIBRARY_OUTPUT_DIRECTORY "${SOURCE_DIR}/src")
-target_link_libraries(myproject ${ZEROMQ_LIBRARIES} ${MORE_LIBRARIES})
-
-install(TARGETS myproject
-    LIBRARY DESTINATION "lib${LIB_SUFFIX}" # .so file
-    ARCHIVE DESTINATION "lib${LIB_SUFFIX}" # .lib file
-    RUNTIME DESTINATION bin              # .dll file
-)
-
-########################################################################
-# pkgconfig
-########################################################################
-set(VERSION "${MYPROJECT_VERSION}")
-set(prefix "${CMAKE_INSTALL_PREFIX}")
-set(exec_prefix "\${prefix}")
-set(libdir "\${prefix}/lib${LIB_SUFFIX}")
-set(includedir "\${prefix}/include")
-configure_file(
-    "${SOURCE_DIR}/src/libmyproject.pc.in"
-    "${SOURCE_DIR}/src/libmyproject.pc"
-@ONLY)
-
-install(
-    FILES "${SOURCE_DIR}/src/libmyproject.pc"
-    DESTINATION "lib${LIB_SUFFIX}/pkgconfig"
-)
 
 ########################################################################
 # executables

--- a/zproject_cmake.gsl
+++ b/zproject_cmake.gsl
@@ -17,6 +17,7 @@ enable_testing()
 
 set(SOURCE_DIR "${CMAKE_CURRENT_SOURCE_DIR}")
 
+.if count (project.class)
 ########################################################################
 # determine version
 ########################################################################
@@ -31,6 +32,7 @@ endforeach(which)
 
 set($(PROJECT.NAME)_VERSION ${$(PROJECT.NAME)_MAJOR_VERSION}.${$(PROJECT.NAME)_MINOR_VERSION}.${$(PROJECT.NAME)_PATCH_VERSION})
 
+.endif
 ########################################################################
 # platform.h
 ########################################################################
@@ -128,6 +130,7 @@ set ($(project.name)_headers
 source_group ("Header Files" FILES ${$(project.name)_headers})
 install(FILES ${$(project.name)_headers} DESTINATION include)
 
+.if count (project.class)
 ########################################################################
 # library
 ########################################################################
@@ -174,6 +177,7 @@ install(
     DESTINATION "lib${LIB_SUFFIX}/pkgconfig"
 )
 
+.endif
 ########################################################################
 # executables
 ########################################################################

